### PR TITLE
TASK: Update PHP coding guidelines

### PR DESCRIPTION
**Pull request by kdambekalns**

### CODE REVIEW

**Comment by kdambekalns** [comment]
@kitsunet This should take care of the things you noticed…

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst [hunk 1]
```diff
@@ -375,113 +370,36 @@ These are the rules for naming files:
...
-    Contains the class ``\TYPO3\Flow\\Tests\Unit\Package\PackageManagerTest`` which
+``Neos.Flow/Tests/Unit/Package/PackageManagerTest.php``
+    Contains the class ``\Neos\Flow\Tests\Unit\Package\PackageManagerTest`` which
     is a PHPUnit testcase for ``Package\PackageManager``.
 
 
 PHP code formatting
 ===================
 
-Strings
```
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: The part on string concatenation must stay, turns out PSR-2 doesn't specify this.

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst [hunk 2]
```diff
@@ -66,35 +66,30 @@ General considerations
 *The Flow standard file header*::
 
  <?php
- namespace YourCompany\Package\Stuff\Here;
+ namespace Neos\Flow\Something\New;
```
Comment (bwaidelich) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Why change this example namespace from "YourCompany\Package" to "Neos\Flow"?

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst [hunk 3]
```diff
@@ -135,8 +130,8 @@ willingly and for historical reasons.
 Vendor namespaces
 -----------------
 
-The base for namespaces as well as package keys is the vendor namespace. Since Flow started
-as part of the TYPO3 project, the core team decided to choose "TYPO3" as our vendor
+The base for namespaces as well as package keys is the vendor namespace. Since Flow is
+part of the Neos project, the core team decided to choose "Neos" as our vendor
 namespace (at some point this will change). The Object Manager for example is known under
 the class name ``TYPO3\Flow\Object\ObjectManager``. In our examples you will find the ``Acme``
```
Comment (bwaidelich) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: should be `Neos\Flow\ObjectManagement\ObjectManager`

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst [hunk 4]
```diff
@@ -158,7 +153,7 @@ only the characters a-z, A-Z, 0-9 and the dash sign "-" are allowed for package
 don't use special characters.
 
 The full package key is then built by combining the vendor namespace and the package,
-like ``TYPO3.Fluid`` or ``Acme.Demo``.
+like ``Neos.Fluid`` or ``Acme.Demo``.
```
Comment (bwaidelich) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: `Neos.Fluid` doesn't exist

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst [hunk 5]
```diff
@@ -576,27 +494,24 @@ The syntax is as follows:
...
-   Introduce xyz service                            // BAD, missing code prefix
-   BUGFIX: Fixed bug xyz                            // BAD, subject should be written in present tense
-   WIP !!! TASK: A breaking change                  // BAD, subject has to start with [!!!] for breaking changes
-   BUGFIX: SessionManager removes expired sessions  // GOOD, the line explains what the change does, not what the
-                                                       bug is about (this should be explained in the following lines
-                                                       and in the related bug tracker ticket)
+   Introduce xyz service                               // BAD, missing code prefix
+   BUGFIX: Fixed bug xyz                               // BAD, subject should be written in present tense
+   WIP !!! TASK: A breaking change                     // BAD, subject has to start with [!!!] for breaking changes
+   BUGFIX: SessionManager will remove expired sessions // GOOD, the line explains what the change does, not what the
```
Comment (bwaidelich) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Why this change??
The former adheres to http://chris.beams.io/posts/git-commit/ the latter doesn't

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst [hunk 6]
```diff
@@ -135,8 +130,8 @@ willingly and for historical reasons.
 Vendor namespaces
 -----------------
 
-The base for namespaces as well as package keys is the vendor namespace. Since Flow started
-as part of the TYPO3 project, the core team decided to choose "TYPO3" as our vendor
+The base for namespaces as well as package keys is the vendor namespace. Since Flow is
+part of the Neos project, the core team decided to choose "Neos" as our vendor
 namespace (at some point this will change). The Object Manager for example is known under
```
Comment (mficzel) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: `(at some point this will change)` in the next line should be removed

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst [hunk 7]
```diff
@@ -375,113 +370,36 @@ These are the rules for naming files:
 * Each file must contain only one class or interface
 * Names of files containing code for unit tests must be the same as the class which is
   tested, appended with "Test.php".
-* Files are placed in a directory structure representing the namespace structure.
+* Files are placed in a directory structure representing the namespace structure. You may
+  use PSR-0 ro PSR-4 autoloading as you like. We generally use PSR-4.
```
Comment (mficzel) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: 'ro' >> 'or'

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst (hunk 2 above)
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Because otherwise the "standard header" doesn't match. We can reverse that, of course.

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst (hunk 6 above)
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Whoops. :)

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst (hunk 3 above)
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Yup.

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst (hunk 4 above)
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Neos.Fluid has left the building…

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst (hunk 7 above)
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Mirror, mirror on the wall…

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst (hunk 5 above)
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Because the former did not really be clear about what the bug or fix is. Was the bug that sessions are removed or the opposite? But the new version is still not really køøl, true.

Comment (bwaidelich) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Just for the record: My point was about the "imparative mood". I think it's a very good tip to form the subject line so that it can complete the phrase "If applied, this commit will ...".

**Comment by bwaidelich** [comment]
@kdambekalns Yihaa, thank you! Good to merge IMO

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst (hunk 5 above)
Comment (kdambekalns) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst]: Ack. That's why it's now

__If applied, this commit will Make SessionManager remove expired sessions__

Thanks for picking the nit!
